MH-13451 Improve delete-event submit button

KatrinIhler opened the pull request:
- Rename button to 'delete' if there is nothing to retract
- Make delete-event button red if event8s) can be deleted immediately (makes users aware, consistent with delete-series)

Commit:
MH-13451 Rename delete-event-modal submit button

Show 'delete' instead of 'retract' if there's nothing to retract

Commit:
MH-13451 Make delete-event button red

Change the color of the submit button in the delete-event-dialog to red
if the event(s) can be deleted immediately to make users aware and to be
consistent with delete-series.

KatrinIhler commented:
@staubesv Just noticed that this doesn't quite work as intended yet, give me a few more minutes! ;)

KatrinIhler commented:
@staubesv Ready for review!

staubesv reviewed (changes requested):
The behaviour is not really consistent:

Case 1:
User selects one or more events without publications.
Button: "Delete" (red)

Case 2:
User selects one or more events with publications.
Button: "Retract" (green)

Case 3:
User selects one or more events without publications and one or more events with publications.
Button: "Retract" (green)

Case 3 is suggesting that no events are deleted.

What do you think about the handling of case 3?

KatrinIhler commented:
@staubesv Good question. I assume in that case some events are deleted and others just retracted? In that case the button should reflect that...

staubesv commented:
@KatrinIhler Yes, both the delete and retract operations will be executed.

KatrinIhler commented:
@staubesv Alright, I'm gonna try to take a quick look at this today.

Commit:
MH-13451 Add third case for delete event button

Make button red and change the text to 'Retract/Delete' if there are
published and unpublished events to delete.

KatrinIhler commented:
@staubesv How about this? Not quite sure about "Retract/Delete" yet, maybe an "or" would be preferable...opinions?

staubesv commented:
@KatrinIhler I suggest "Retract & Delete" as this label is only shown if both retractions and deletions will be performed.

KatrinIhler commented:
@staubesv I think that wording is confusing because users could assume that opencast retracts the published events and then deletes all events, which...would make sense, but is not what is happening.
